src/Workspaces/Core/Portable/PatternMatching/PatternMatcher.cs

```diff
@@ -52,7 +58,20 @@ internal abstract partial class PatternMatcher : IDisposable
             bool allowFuzzyMatching = false)
         {
             culture = culture ?? CultureInfo.CurrentCulture;
+            if (culture.Name == "en-US")
```

Comment:
Sorry, i'm not seeing what the concern is precisely.  What is the troubling bit?  All i'm trying to do is optimize the common case, while falling back in less common cases.  There may be different results, but it seems like:

1. there would be practically no chance of hitting this.
2. if you hit it, so what?  you'd get a slightly less optimal result.

This is not compiler code.  This is code used for fast searching a solution for matching hits.  In that regard it's similar to a search engine, and this is a tiny tweak to try to speed up the 99.9% case, while probably not ever being noticeable.